### DISCUSSION TITLE
Auth user against host

### DISCUSSION THREAD

**Discussion by lab-at-nohl** [immich-app/immich#13433] [Q&A] [answered]
Hello, 

I know immich internal user/auth mechanism as well as OAuth. However, it seems logically to me to auth against **the host's user accounts** (i.e. passwd/shadow). Even more precisely, it would be handy to limit access to users in a specific group...

How does NAS integrations do it? Do they really require to have a dedicated user scheme for "only" immich?

However, the app is really nice!

Best
Johannes

**Comment by mmomjian** [comment]
Thanks for the request. It’s very unlikely that we’ll be adding this. Docker containers don’t / shouldn’t have access to this information. Maybe there is an oauth integration that has what you’re looking for.

**Reply by lab-at-nohl** [reply]
Thanks for the swift reply. I get your point. However, OAuth2 requires another web service (anywhere but probably on the storage machine). For a consistent NAS setup anything internal is required, though. I further researched on possibilities; The only reasonable choice for a preconfigured container is a full appliance like keycloak. 

Would you consider to **authenticate against SMB** (v2 or v3)? The server might be Windows Server but, for a NAS, it would be likely SAMBA. I mean without mounting any shares (which may be added later), just the auth part. SAMBA comes with its own user password mechanism. Also, this way even a change of password could be delivered. That would make immich the center of user self-management on the NAS. 

For a simple wrapper of the linux `smbclient` see https://www.npmjs.com/package/samba-client. Also a native implementation exist, https://github.com/Node-SMB/marsaud-smb2; But I doubt it can change passwords. `smbclient` could be included easily into your bookworm image. 

I see that people are kernel mounting smb shares inside your image, too. However the client might be a better option.

**Reply by bo0tzz** [reply, marked as answer]
> Would you consider to authenticate against SMB (v2 or v3)? 

No. The built in authentication is deliberately as minimal as possible. For any more complex need, use oauth.
